mempool/mempool.go

```diff
@@ -0,0 +1,395 @@
+package mempool
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/holiman/uint256"
+
+	"github.com/cosmos/evm/mempool/miner"
+	"github.com/cosmos/evm/mempool/txpool"
+	"github.com/cosmos/evm/mempool/txpool/legacypool"
+	"github.com/cosmos/evm/x/precisebank/types"
+	evmtypes "github.com/cosmos/evm/x/vm/types"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sdkmempool "github.com/cosmos/cosmos-sdk/types/mempool"
+)
+
+var _ sdkmempool.ExtMempool = &EVMMempool{}
+
+type (
+	// EVMMempool is a unified mempool that manages both EVM and Cosmos SDK transactions.
+	// It provides a single interface for transaction insertion, selection, and removal while
+	// maintaining separate pools for EVM and Cosmos transactions. The mempool handles
+	// fee-based transaction prioritization and manages nonce sequencing for EVM transactions.
+	EVMMempool struct {
```

Comment:
I think it most likely makes sense to add a logger to this.  Right now we really don't have any logging from this new mempool logic and it could be useful to add debug logs to make our lives easier in the future. Happy to just have this as a follow-up task